### PR TITLE
Av-embeds `externalEmbedUrl` logic refactor

### PR DESCRIPTION
**Pull request by amoore108**
Overall changes
======
Refactors the method of constructing the `externalEmbedUrl` value for the media player, which is also used for the JSON+LD `embedUrl` value.

This should properly extract and set these values and append correct media identifier values (pid/vpid) where necessary.

Code changes
======
- Adds new `getExternalEmbedUrl` function to have code separation when generating the `externalEmbedUrl` and `embedUrl` value
- Adds `getAmpIframeUrl` which is not yet used, but will be needed to generate the correct /ws/av-embeds routes when we uplift/migrate the AMP iframe over to the new media loader component

Testing
======
1. _List the steps used to test this PR._

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

File: src/app/components/MediaLoader/utils/urlConstructors.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+const getAmpBaseUrl = () => {
+  switch (true) {
+    case isLive():
+      return LIVE_AMP_URL;
+    case isDev():
+      return DEV_AMP_URL;
+    default:
+      return TEST_AMP_URL;
+  }
+};
```
Comment (karinathomasbbc) [inline src/app/components/MediaLoader/utils/urlConstructors.ts]: Could we simplify this - looks like we don't need a switch/case statement, but a simple if/ternary statement. We can get rid of all references to dev & just keep test & live. 

```suggestion

const getAmpBaseUrl = () => {
  return isLive() ? LIVE_AMP_URL : TEST_AMP_URL;
};
```

Comment (karinathomasbbc) [reply src/app/components/MediaLoader/utils/urlConstructors.ts]: In fact, we could delete this function, and simply do this on line 34

File: src/app/components/MediaLoader/utils/urlConstructors.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+type FuncProps = {
+  id: string;
+  versionID?: string;
+  lang?: string;
+};
+
+export const getAmpIframeUrl = ({ id, versionID, lang }: FuncProps) => {
+  const { platform, service, variant, assetId } = parseAvRoute(id);
+
+  const ampBaseUrl = getAmpBaseUrl();
```
Comment (karinathomasbbc) [inline src/app/components/MediaLoader/utils/urlConstructors.ts]:
```suggestion
  const ampBaseUrl = isLive() ? LIVE_AMP_URL : TEST_AMP_URL;
```

Comment (karinathomasbbc) [reply src/app/components/MediaLoader/utils/urlConstructors.ts]: See comment above RE deletion / simplification

File: src/app/components/MediaLoader/utils/urlConstructors.ts
Comment (karinathomasbbc) [inline src/app/components/MediaLoader/utils/urlConstructors.ts:1]: This is very similar to #lib/utilities/getUrlHelpers/getEmbedUrl - can we get rid of `getEmbedUrl` at some point?

Comment (amoore108) [reply src/app/components/MediaLoader/utils/urlConstructors.ts:1]: Yes we'll get rid of it when we move the Amp-iframe over to the new Media Loader. Its only really used there now.

File: src/app/components/MediaLoader/utils/urlConstructors.ts (hunk 2 above)
Comment (amoore108) [reply src/app/components/MediaLoader/utils/urlConstructors.ts]: Nice, yep thats much better.

File: src/app/components/MediaLoader/utils/urlConstructors.ts
Comment (karinathomasbbc) [reply src/app/components/MediaLoader/utils/urlConstructors.ts:1]: Cool, thanks!

File: src/app/components/MediaLoader/utils/buildSettings.test.ts [hunk 3]
```diff
@@ -167,7 +167,7 @@ describe('buildSettings', () => {
           },
           enableToucan: true,
           externalEmbedUrl:
-            '/ws/av-embeds/cps/serbian/lat/srbija-68707945/p01k6msp/sr-latn',
+            'https://www.test.bbc.com/serbian/lat/av-embeds/srbija-68707945/vpid/p01k6msp',
```
Comment (amoore108) [inline src/app/components/MediaLoader/utils/buildSettings.test.ts:170]: These are more accurate in my mind. `/ws/av-embeds` should only be for our AMP pages when it comes to CPS assets. The `<service>/av-embeds/` route is used for external embedding, so this feels better.

File: src/app/components/MediaLoader/utils/urlConstructors.ts
Comment (amoore108) [reply src/app/components/MediaLoader/utils/urlConstructors.ts:1]: Should probably do the AMP stuff after the deadline. Just don't want to make too many changes at this stage!

File: src/app/components/MediaLoader/utils/urlConstructors.test.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        expected:
+          'https://www.test.bbc.com/ws/av-embeds/articles/cd1rmn075d1o/p0jd37n8/ig',
+      },
+      {
+        description: 'Optimo Syndicated route (no versionID)',
+        id: 'ws/av-embeds/articles/cd1rmn075d1o/p0jd37n8/ig',
+        lang: 'ig',
+        expected:
+          'https://www.test.bbc.com/ws/av-embeds/articles/cd1rmn075d1o/ig',
+      },
```
Comment (andrewscfc) [inline src/app/components/MediaLoader/utils/urlConstructors.test.ts:147]: Really nice to see the requirements clearly illustrated here 🥳

**Review by andrewscfc — APPROVED** [review]
Looks great to me 🧢

**Review by LilyL0u — APPROVED** [review]

**Pull request merged**